pkg/api/componentreadiness/querygenerators.go

```diff
@@ -311,7 +340,184 @@ func (f *fallbackTestQueryGenerator) getFallbackBaseQueryStatus(ctx context.Cont
+					SELECT
+						ANY_VALUE(test_name) AS test_name,
+						ANY_VALUE(testsuite) AS test_suite,
+						cm.id as test_id,
+						%s
+						COUNT(cm.id) AS total_count,
+						SUM(adjusted_success_val) AS success_count,
+						SUM(adjusted_flake_count) AS flake_count,
+						MAX(CASE WHEN adjusted_success_val = 0 THEN modified_time ELSE NULL END) AS last_failure,
+						ANY_VALUE(cm.component) AS component,
+						ANY_VALUE(cm.capabilities) AS capabilities,
+					FROM (%s)
+					INNER JOIN latest_component_mapping cm ON testsuite = cm.suite AND test_name = cm.name
+`,
+		c.client.Dataset, c.client.Dataset, selectVariants, fmt.Sprintf(dedupedJunitTable, jobNameQueryPortion, c.client.Dataset, junitTable, c.client.Dataset))
+
+	queryString += joinVariants
+
+	groupString := fmt.Sprintf(`
+					GROUP BY
+						%s
+						cm.id `, groupByVariants)
+
+	queryString += `WHERE cm.staff_approved_obsolete = false AND 
+						(variant_registry_job_name LIKE 'periodic-%%' OR variant_registry_job_name LIKE 'release-%%' OR variant_registry_job_name LIKE 'aggregator-%%') 
+						AND NOT REGEXP_CONTAINS(variant_registry_job_name, @IgnoredJobs)`
+
+	commonParams := []bigquery.QueryParameter{
+		{
+			Name:  "IgnoredJobs",
+			Value: ignoredJobsRegexp,
+		},
+	}
+	if c.IgnoreDisruption {
+		queryString += ` AND NOT 'Disruption' in UNNEST(capabilities)`
+	}
+
+	// fallback queries get all variants with no filtering
+	// so all tests are fetched then cached
+	if !isFallback {
+		variantGroups := c.IncludeVariants
+		// potentially cross-compare variants for the sample
+		if isSample && len(c.VariantCrossCompare) > 0 {
+			variantGroups = c.CompareVariants
+		}
+		if variantGroups == nil { // server-side view definitions may omit a variants map
+			variantGroups = map[string][]string{}
+		}
+
+		for _, group := range sortedKeys(variantGroups) {
+			variants := variantGroups[group]
+			queryString += " AND ("
+			first := true
+			for _, variant := range variants {
+				if first {
+					queryString += fmt.Sprintf(`jv_%s.variant_value = '%s'`, group, variant)
+					first = false
+				} else {
+					queryString += fmt.Sprintf(` OR jv_%s.variant_value = '%s'`, group, variant)
+				}
+			}
+			queryString += ")"
+		}
+
+		for _, group := range sortedKeys(c.RequestedVariants) {
+			variant := c.RequestedVariants[group]
+			queryString += fmt.Sprintf(` AND jv_%s.variant_value = '%s'`, group, variant)
+		}
+		if c.Capability != "" {
+			queryString += " AND @Capability in UNNEST(capabilities)"
+			commonParams = append(commonParams, bigquery.QueryParameter{
+				Name:  "Capability",
+				Value: c.Capability,
+			})
+		}
+		if c.TestID != "" {
+			queryString += ` AND cm.id = @TestId`
+			commonParams = append(commonParams, bigquery.QueryParameter{
+				Name:  "TestId",
+				Value: c.TestID,
+			})
+		}
+	}
+	return queryString, groupString, commonParams
+}
+
+func fetchTestStatus(ctx context.Context, query *bigquery.Query) (map[string]crtype.TestStatus, []error) {
```

Comment:
`fetchTestStatus` is confusingly close to `queryTestStatus`. i might like something to indicate that it's just executing the query and returning results, like `fetchTestStatusResults`

`fetchJobRunTestStatus` is already distinct enough but for consistency with the above might be `fetchJobRunTestStatusResults`